docs/guides/assets.md

```diff
@@ -242,9 +246,29 @@ Calling this API will set the asset base path for all Stencil components attache
 As a result, calling `setAssetPath` should not be done from within a component in order to prevent unwanted side effects
 when using a component.
 
-If the file calling `setAssetPath` is a module, it's recommended to use `import.meta.url`.
+If the file calling `setAssetPath` is a module and your compiler is able to resolve the file location at build time, based on where the application is being deployed, you can use `import.meta.url`. In most cases however we recommend to use:
```

Comment:
I'm a little confused by this part:

> your compiler is able to resolve the file location at build time

I thought this was a runtime construct/action, which we seem to imply this above this line with:

> Calling this API will set the asset base path for all Stencil components attached to a Stencil runtime.
As a result, calling `setAssetPath` should not be done from within a component in order to prevent unwanted side effects
when using a component.

Do I have that right? Or is there some build time actions we do that I don't recall (equally possible 😆)

Reply:
I am referring here to e.g. an external compiler that a user is using to create application bundle. In hindsight this seems indeed very ambiguous, I will try to rephrase this.